protocols/http.go

```diff
@@ -79,7 +81,7 @@ func (c *httpClient) Close() error { return nil }
 
 func (c *httpClient) GetID() string { return c.id }
 
-func (c *httpClient) Send(req *pb.TheRequest) (*pb.TheResponse, error) {
+func (c *httpClient) Send(_ context.Context, req *pb.TheRequest) (*pb.TheResponse, error) {
```

Comment:
For parity with the gRPC implementation of this method, I recommend passing this context through to the POST request that this method is making. You could do something like:

```diff
diff --git a/protocols/http.go b/protocols/http.go
index 8a56a0e..0ea423a 100644
--- a/protocols/http.go
+++ b/protocols/http.go
@@ -81,13 +81,18 @@ func (c *httpClient) Close() error { return nil }
 
 func (c *httpClient) GetID() string { return c.id }
 
-func (c *httpClient) Send(_ context.Context, req *pb.TheRequest) (*pb.TheResponse, error) {
+func (c *httpClient) Send(ctx context.Context, req *pb.TheRequest) (*pb.TheResponse, error) {
 	json, err := marshallProtobufToJSON(req)
 	if err != nil {
 		return nil, err
 	}
 
-	resp, err := c.clientForDownsteamServers.Post(c.serverURL, "application/json", strings.NewReader(json))
+	httpReq, err := http.NewRequest("POST", c.serverURL, strings.NewReader(json))
+	if err != nil {
+		return nil, err
+	}
+	httpReq.Header.Set("Content-Type", "application/json")
+	resp, err := c.clientForDownsteamServers.Do(httpReq.WithContext(ctx))
 	if err != nil {
 		return nil, err
 	}
```